### PR TITLE
Modernise remaining uses of `GetSortedLanes()`

### PR DESCRIPTION
**Pull request by aubergine10**
Final batch of `GetSortedLanes()` updates.

Also reduced memalloc/gc of lists in some `Record` classes, and removed some unused `using` statements.

### CODE REVIEW

File: TLM/TLM/Util/Record/LaneArrowsRecord.cs [hunk 1]
```diff
@@ -31,24 +29,32 @@ public class LaneArrowsRecord : IRecordable {
...
                 sort: false);
-            foreach (var lane in lanes) {
-                LaneArrowsRecord laneData = new LaneArrowsRecord {
-                    LaneId = lane.laneId,
-                };
-                ret.Add(laneData);
-            }
-            ret.TrimExcess();
+
+            var ret = new List<LaneArrowsRecord>(lanes.Count);
```
Comment (kvakvs) [inline TLM/TLM/Util/Record/LaneArrowsRecord.cs:53]: Can this be zero size? Return a premade static zero size readonly list in such case, do not allocate.

File: TLM/TLM/Util/Record/SpeedLimitLaneRecord.cs [hunk 2]
```diff
@@ -43,29 +42,35 @@ public class SpeedLimitLaneRecord : IRecordable {
...
-                segmentId,
-                ref segmentId.ToSegment(),
+
+            var lanes = segmentId.ToSegment().GetSortedLanes(
                 null,
                 LANE_TYPES,
                 VEHICLE_TYPES,
                 sort: false);
+
+            var ret = new List<SpeedLimitLaneRecord>(lanes.Count);
```
Comment (kvakvs) [inline TLM/TLM/Util/Record/SpeedLimitLaneRecord.cs:62]: Same comment here, can this be zero size? If so, return a premade zero size static readonly list.

File: TLM/TLM/Util/Record/VehicleRestrictionsLaneRecord.cs [hunk 3]
```diff
@@ -53,26 +52,33 @@ public class VehicleRestrictionsLaneRecord : IRecordable {
...
-                segmentId,
-                ref segmentId.ToSegment(),
+
+            var lanes = segmentId.ToSegment().GetSortedLanes(
                 null,
                 LANE_TYPES,
                 VEHICLE_TYPES,
                 sort: false);
+
+            var ret = new List<VehicleRestrictionsLaneRecord>(lanes.Count);
```
Comment (kvakvs) [inline TLM/TLM/Util/Record/VehicleRestrictionsLaneRecord.cs:72]: Can this be zero size, does this happen often? If so, return a premade zero size static readonly list.

**Review by kvakvs — APPROVED** [review]

File: TLM/TLM/Util/Record/SpeedLimitLaneRecord.cs (hunk 2 above)
Comment (kianzarrin) [reply TLM/TLM/Util/Record/SpeedLimitLaneRecord.cs:62]: > If so, return a premade zero size static readonly list.

That would be `Enumerable.Empty<SpeedLimitRecord>()`

File: TLM/TLM/Util/Record/LaneArrowsRecord.cs (hunk 1 above)
Comment (kianzarrin) [reply TLM/TLM/Util/Record/LaneArrowsRecord.cs:53]: it does need the space.

File: TLM/TLM/Util/Record/SpeedLimitLaneRecord.cs (hunk 2 above)
Comment (kianzarrin) [reply TLM/TLM/Util/Record/SpeedLimitLaneRecord.cs:62]: wait that does not return a list

**Comment by aubergine10** [comment]
@kianzarrin how often do the record lists get created? I assume it's only in asset editor and only when plopping assets on map so not performance critical?

**Comment by kianzarrin** [comment]
Not performance critical. It happens due to a user event:
- asset editor save
- move it copy/save
- TMPE priority road button pressed (so that I can undo)
- in future: every time user changes TMPE rules (so that I can undo)

**Comment by aubergine10** [comment]
bump

**Review by krzychu124 — APPROVED** [review]

**Review by kianzarrin — APPROVED** [review]

**Pull request merged**